cppcache/src/Exception.cpp

```diff
@@ -36,90 +34,29 @@ void Exception::setStackTraces(bool stackTraceEnabled) {
   s_exceptionStackTraceEnabled = stackTraceEnabled;
 }
 
-Exception::Exception(const char* msg1, const char* msg2, bool forceTrace,
-                     const std::shared_ptr<Exception>& cause)
-    : m_stack(), m_cause(cause) {
-  size_t len1 = 0;
-  if (msg1) {
-    len1 = strlen(msg1);
-  }
-  size_t len2 = 0;
-  if (msg2) {
-    len2 = strlen(msg2);
-  }
-  size_t len = len1 + len2;
-  char* msg;
-  GF_NEW(msg, char[len + 1]);
-  if (msg1) {
-    ACE_OS::memcpy(msg, msg1, len1);
-  }
-  if (msg2) {
-    ACE_OS::memcpy(msg + len1, msg2, len2);
-  }
-  msg[len] = '\0';
+Exception::Exception(const std::string& msg)
+  : Exception(msg.c_str()) {
+}
 
-  if (s_exceptionStackTraceEnabled || forceTrace) {
-    m_stack = std::make_shared<StackTrace>();
+Exception::Exception(const char* msg1)
+  : message(msg1) {
+  if (s_exceptionStackTraceEnabled/* || forceTrace*/) {
+    m_stack = std::unique_ptr<StackTrace>();
   }
-  m_message = CacheableString::createNoCopy(msg, static_cast<int32_t>(len));
 }
 
-Exception::Exception(const std::string& msg1) : Exception(msg1.c_str()) {}
+const char *Exception::what() const noexcept {
+  return message.c_str();
+}
 
-Exception::~Exception() {}
+Exception::~Exception() noexcept {}
 
 const char _exception_name_Exception[] = "apache::geode::client::Exception";
 
 const char* Exception::getName() const { return _exception_name_Exception; }
```

Comment:
Is getName is ours so why not std::string?

Reply:
Yes.  No reason really.  I just figured the std::string story would cover it later.  ;)

Reply:
Way to pass the buck!